typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5ImpersonatedAccountTxSubmitter.ts

```diff
@@ -25,19 +20,19 @@ export class EV5ImpersonatedAccountTxSubmitter extends EV5JsonRpcTxSubmitter {
   });
 
   constructor(
-    public readonly multiProvider: MultiProvider,
-    public readonly chain: ChainName,
+    multiProvider: MultiProvider,
     public readonly props: EV5ImpersonatedAccountTxSubmitterProps,
   ) {
-    super(multiProvider, chain);
+    super(multiProvider);
   }
 
   public async submit(
     ...txs: PopulatedTransaction[]
   ): Promise<TransactionReceipt[]> {
-    const impersonatedAccount = await impersonateAccount(this.props.address);
-    this.multiProvider.setSigner(this.chain, impersonatedAccount);
-    super.multiProvider.setSigner(this.chain, impersonatedAccount);
+    const impersonatedAccount = await impersonateAccount(
+      this.props.userAddress,
+    );
+    super.multiProvider.setSharedSigner(impersonatedAccount);
     return await super.submit(...txs);
```

Comment:
idk what the implication is of not doing this but should we 
`stopImpersonating` after submitting?

Reply:
Interesting- I think we should do this in order to keep the submission atomic, and I don't think this should have any downstream effects since we are also impersonating during this call

will address as a part of https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3786